### PR TITLE
Make pytest.mark.asyncio compatible with classic functions

### LINKED ISSUE #109: Marker pytest.mark.asyncio should be compatible with classic functions also
Using module-global marker `pytestmark = pytest.mark.asyncio` is an elegant way to achieve https://github.com/pytest-dev/pytest-asyncio/issues/61. However, it currently fails if the module has `async def` and classic `def` tests intermixed:

````py
import pytest
import asyncio

pytestmark = pytest.mark.asyncio

def test_classic():
    assert bool(1) == True

async def test_async():
    await asyncio.sleep(0)
    assert bool(0) == False
````
````pytb
============================= test session starts =============================
platform win32 -- Python 3.6.1, pytest-4.2.0, py-1.7.0, pluggy-0.8.1
rootdir: C:\temp, inifile:
plugins: asyncio-0.10.0
collected 2 items

asyncio-test.py F.                                                       [100%]

================================== FAILURES ===================================
________________________________ test_classic _________________________________

pyfuncitem = <Function test_classic>

    @pytest.mark.tryfirst
    def pytest_pyfunc_call(pyfuncitem):
        """
        Run asyncio marked test functions in an event loop instead of a normal
        function call.
        """
        for marker_name, fixture_name in _markers_2_fixtures.items():
            if marker_name in pyfuncitem.keywords \
                    and not getattr(pyfuncitem.obj, 'is_hypothesis_test', False):
                event_loop = pyfuncitem.funcargs[fixture_name]
    
                funcargs = pyfuncitem.funcargs
                testargs = {arg: funcargs[arg]
                            for arg in pyfuncitem._fixtureinfo.argnames}
    
                event_loop.run_until_complete(
                    asyncio.ensure_future(
>                       pyfuncitem.obj(**testargs), loop=event_loop))

C:\ProgramData\Anaconda3\lib\site-packages\pytest_asyncio\plugin.py:158: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

coro_or_future = None

    def ensure_future(coro_or_future, *, loop=None):
        """Wrap a coroutine or an awaitable in a future.
    
        If the argument is a Future, it is returned directly.
        """
        if futures.isfuture(coro_or_future):
            if loop is not None and loop is not coro_or_future._loop:
                raise ValueError('loop argument must agree with Future')
            return coro_or_future
        elif coroutines.iscoroutine(coro_or_future):
            if loop is None:
                loop = events.get_event_loop()
            task = loop.create_task(coro_or_future)
            if task._source_traceback:
                del task._source_traceback[-1]
            return task
        elif compat.PY35 and inspect.isawaitable(coro_or_future):
            return ensure_future(_wrap_awaitable(coro_or_future), loop=loop)
        else:
>           raise TypeError('A Future, a coroutine or an awaitable is required')
E           TypeError: A Future, a coroutine or an awaitable is required

C:\ProgramData\Anaconda3\lib\asyncio\tasks.py:520: TypeError
===================== 1 failed, 1 passed in 0.29 seconds ======================
````

It is proposed that `pytest.mark.asyncio` is compatible with classic tests also (being a no-op there).

### PR DESCRIPTION
**Pull request by Tadaboody**
Closes #109

### CODE REVIEW

File: tests/test_simple_35.py [hunk 1]
```diff
@@ -72,6 +72,10 @@ def test_unused_port_factory_fixture(unused_tcp_port_factory, event_loop):
     event_loop.stop()
     event_loop.close()
 
+@pytest.mark.asyncio
+def test_non_corutine_marked():
+    """Tests that marked non-corutines function normally (Issue #109)"""
```
Comment (thilp) [inline tests/test_simple_35.py]:
```suggestion
    """Tests that marked non-coroutines function normally (Issue #109)"""
```

**Comment by blueyed** [comment]
LGTM, please rebase and take the suggested doc fix.

**Comment by mitzkia** [comment]
@Tadaboody could you please update this PR (rebase and doc update). It would help me a lot.

**Comment by Tadaboody** [comment]
Sorry this took so long 😅 wasn't really maintaining my github

**Comment by Tadaboody** [comment]
It looks like currently there isn't a problem with non-coroutines I deleted my old change (it had problems with hypothesis) and the test I added still passed. Is it enough to say that the issue is resolved?

**Pull request closed without merging**